--- ./eProsima/ROS-RMW-Fast-RTPS-cpp/rmw_fastrtps_cpp/src/functions.cpp
+++ ./eProsima/ROS-RMW-Fast-RTPS-cpp/rmw_fastrtps_cpp/src/functions.cpp
@@ -277,10 +277,9 @@ extern "C"
             return NULL;
         }
 
-        eprosima::Log::setVerbosity(eprosima::VERB_ERROR);
-
         ParticipantAttributes participantParam;
         participantParam.rtps.builtin.domainId = static_cast<uint32_t>(domain_id);
+        participantParam.rtps.builtin.leaseDuration_announcementperiod.seconds = 5;
         participantParam.rtps.setName(name);
 
         Participant *participant = Domain::createParticipant(participantParam);
