--- ./ros2/rmw_opensplice/rmw_opensplice_cpp/src/qos.cpp
+++ ./ros2/rmw_opensplice/rmw_opensplice_cpp/src/qos.cpp
@@ -56,7 +56,7 @@ bool set_entity_qos_from_profile(const rmw_qos_profile_t & qos_profile,
 
   switch (qos_profile.durability) {
     case RMW_QOS_POLICY_TRANSIENT_LOCAL_DURABILITY:
-      entity_qos.durability.kind = DDS::TRANSIENT_LOCAL_DURABILITY_QOS;
+      entity_qos.durability.kind = DDS::TRANSIENT_DURABILITY_QOS;
       break;
     case RMW_QOS_POLICY_VOLATILE_DURABILITY:
       entity_qos.durability.kind = DDS::VOLATILE_DURABILITY_QOS;
